src/clj/rems/api/services/resource.clj

```diff
@@ -1,18 +1,31 @@
 (ns rems.api.services.resource
-  (:require [rems.api.services.util :as util]
+  (:require [com.rpl.specter :refer [ALL transform]]
+            [rems.api.services.util :as util]
             [rems.db.catalogue :as catalogue]
             [rems.db.core :as db]
             [rems.db.licenses :as licenses]
+            [rems.db.organizations :as organizations]
             [rems.db.resource :as resource])
   (:import (org.postgresql.util PSQLException)))
 
-(defn get-resource [id] (resource/get-resource id))
-(defn get-resources [filters] (resource/get-resources filters))
+(defn- join-dependencies [resource]
+  (->> resource
+       organizations/join-organization
+       licenses/join-resource-licenses
+       (transform [:licenses ALL] organizations/join-organization)))
```

Comment:
hmm it kinda feels like licenses should join their own organizations. would that cause some dependency problems?

Reply:
I'm thinking we'll figure this out for the #2179.  But quickly, the db layer license can't join its organization unless the db layer can depend on other db layer namespaces and that leads to the cyclic dependency. Service layer therefore must explicitly join all the related entities that it wants. Here a more automatic dependency resolution would help. Also it's possible the join keys are not always knowable by the lower levels. I experimented with some logic in there but I'd like to clean these up in the #2179 tasks.

Reply:
fair enough